Remove shipment ship event

mbj opened the pull request.

dkubb commented on core/app/models/spree/shipment.rb:
```diff
@@ -117,7 +110,7 @@ def determine_state(order)
       return 'pending' if inventory_units.any? &:backordered?
       return 'shipped' if state == 'shipped'
-      order.paid? || Spree::Config[:auto_capture_on_dispatch] ? 'ready' : 'pending'
+      order.paid? ? 'ready' : 'pending'
```
For consistency, what do you think about writing this method as:

```ruby
def determine_state(order)
  return 'canceled' if order.canceled?
  return 'pending'  unless order.can_ship?
  return 'pending'  if inventory_units.any?(&:backordered?)
  return 'shipped'  if state.eql?('shipped')
  return 'pending'  unless order.paid?
  'ready'
end
```

I realize it's one more line of code, but it makes it more clear that the fall-through condition is `'ready'`. Alternatively, this could be written as an if/elsif/else block and be arguably eaiser to read.

mbj replied:
I'd actually like it more when a more "defenisive" state like `'pending'` would be the default. So lets go for `return 'ready' if order.paid?`

dkubb replied:
So you're thinking something like:

```ruby
def determine_state(order)
  return 'canceled' if order.canceled?
  return 'pending'  unless order.can_ship?
  return 'pending'  if inventory_units.any?(&:backordered?)
  return 'shipped'  if shipped?
  return 'ready'    if order.paid?
  'pending'
end
```

(Note that `state.eql?('shipped')` can become `shipped?` if desired)

mbj replied:
Yeah, I think that is better. All exceptional states are explicitly handled, and the default fall through is save.

mbj replied:
OT: we can remove this mid term and just reflect the order state ;)

dkubb commented:
@mbj can you explain what functionality this removes specifically? Is it something we don't use at all?

mbj commented:
@dkubb It removes the support from core to trigger shipment state change via the `ship` event and adjacent. We removed the UI for triggering this transition in https://github.com/MountainRoseHerbs/spree/pull/263.

The idea is that shipments are right now de-facto used for rate lookup, so we can remove any functionality that is provided on top of that. Than refactor the shipment internals (next step) to not rely on inventory units as we do not need per stock location tracking.

But this change was too if where included, so I removed features add a lower granularity to keep it reviewable.

dkubb commented:
@mbj I will merge this when the build is green.
